### PR TITLE
Circuit cutting: Tidy up documentation

### PR DESCRIPTION
**Pull request by trbromley**
**Context:**

Tidies up the documentation for the new circuit cutting feature

### CODE REVIEW

File: doc/code/qml_transforms_qcut.rst [hunk 1]
```diff
@@ -0,0 +1,32 @@
+:orphan:
+
+qml.transforms.qcut
+===================
```
Comment (josh146) [inline doc/code/qml_transforms_qcut.rst:4]: Hey Tom, you could put this content in the `qcut/__init__.py` module docstring, and avoid needing to have a separate file in `doc/code`.

The advantage here is that is closer to the code 🙂 That way anyone who modifies the `qcut` module can more easily see that they have to update this.

Similar to how the devices subpackage works: https://github.com/PennyLaneAI/pennylane/blob/1bca151b7f3172237654e18fcf42749c9aa5d149/pennylane/devices/__init__.py#L15-L32

**Comment by trbromley** [comment]
[sc-14726]

File: pennylane/transforms/qcut.py [hunk 2]
```diff
@@ -905,21 +935,28 @@ def cut_circuit(
...
     Args:
-        tape (QuantumTape): The tape of the full circuit to be cut.
+        tape (QuantumTape): the tape of the full circuit to be cut
         use_opt_einsum (bool): Determines whether to use the
             `opt_einsum <https://dgasmith.github.io/opt_einsum/>`__ package. This package is useful
             for faster tensor contractions of large networks but must be installed separately using,
             e.g., ``pip install opt_einsum``. Both settings for ``use_opt_einsum`` result in a
             differentiable contraction.
-        device_wires (.wires.Wires): Wires of the device that the cut circuits are to be run on
+        device_wires (Wires): Wires of the device that the cut circuits are to be run on
```
Comment (trbromley) [inline pennylane/transforms/qcut.py:959]:
```suggestion
        device_wires (Wires): wires of the device that the cut circuits are to be run on
```

**Review by anthayes92 — COMMENTED** [review]
Just left comments for now as I'm not sure if more changes are anticipated in this PR e.g updating the changelog entry for the full circuit cutting pipeline? 

Feels like the addition of the `qml_transforms_qcut.rst` file goes against convention for transforms, but then maybe this is a big transform so potentially needs special treatment! However I did find it tricky to navigate to in the docs.

File: doc/code/qml_transforms_qcut.rst [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+This module provides support for quantum circuit cutting, allowing larger circuits to be split into
+smaller circuits that are compatible with devices that have a restricted number of qubits. The main
```
[the remark this answers is not in the document]
Comment (trbromley) [reply doc/code/qml_transforms_qcut.rst]: Thanks! I mentioned `qml.WireCut` here: e3250d0c20b5be1dbb8a4a3669bec9a45db50773

> Is this the right place to go into a little more detail on how the circuits are executed and post processed?

Good idea. We do mention that info in the usage details section of `cut_circuit`, so I moved that here: b791416bebe974b311dfb913c958293ec8e4e5c5

File: doc/code/qml_transforms_qcut.rst [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+batch transform.
+
+.. currentmodule:: pennylane
+
+.. autosummary::
+    :toctree: api
+
+    cut_circuit
+
+Advanced users also have the option to work directly with a :class:`~.QuantumTape` and manipulate
```
[the remark this answers is not in the document]
Comment (trbromley) [reply doc/code/qml_transforms_qcut.rst:22]: I think it's best to leave mentioning automatic functionality until it's more developed. `CutStrategy` already exists, but there is a note added there saying it's WIP. But I agree once we add more automatic functionality we can emphasize more here and perhaps have two `.. autosummary::` lists.

**Review by trbromley — COMMENTED** [review]
Thanks @anthayes92!

> Just left comments for now as I'm not sure if more changes are anticipated in this PR e.g updating the changelog entry for the full circuit cutting pipeline?

No more changes, we can do the changelog in a separate PR.

> Feels like the addition of the qml_transforms_qcut.rst file goes against convention for transforms, but then maybe this is a big transform so potentially needs special treatment!

Yes, I think it's polluting `qml.transforms` too much and better to factor into a separate page.

> However I did find it tricky to navigate to in the docs.

Right! Have partially addressed that here: 5403485f1243487077109c805862957ef2160bfc. Any suggestions to emphasize more would be great! Though also I don't think it needs to be a particularly important page.

File: doc/code/qml_transforms_qcut.rst [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+     1: ──X────────╰C──╰┤ ⟨Z ⊗ Z⟩
+
+     0: ──RX(0.3)──╭Z──╭┤ ⟨Z ⊗ Z⟩
+     1: ──H────────╰C──╰┤ ⟨Z ⊗ Z⟩
+
+     0: ──RX(0.3)─────╭Z──╭┤ ⟨Z ⊗ Z⟩
+     1: ──H────────S──╰C──╰┤ ⟨Z ⊗ Z⟩
+
+The last step is to execute the tapes and postprocess the results using
+:func:`~.qcut_processing_fn`, which converts the results into a tensor network contraction.
```
Comment (anthayes92) [inline doc/code/qml_transforms_qcut.rst:155]: Wording is a bit strange (at least IMO).  Perhaps use:
"which processes the results to the original full circuit output vie a tensor network contraction"

**Review by anthayes92 — APPROVED** [review]
Thanks @trbromley, left a few minor suggestions. Great to see this all tidy and polished :raised_hands: :sparkles: :heavy_check_mark:

File: doc/code/qml_transforms_qcut.rst [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+Our next step is to remove the :class:`~.WireCut` nodes in the graph and replace with
+:class:`~.MeasureNode` and :class:`~.PrepareNode` pairs.
+
+>>> qml.transforms.qcut.replace_wire_cut_nodes(graph)
+
+The :class:`~.MeasureNode` and :class:`~.PrepareNode` pairs are placeholder operations that
+allow us to cut the circuit graph and then iterate over measurement and preparation
+configurations at cut locations. First, the :func:`~.fragment_graph` function pulls apart
+the graph into disconnected components as well as returning the ``communication_graph``
```
[the remark this answers is not in the document]
Comment (trbromley) [reply doc/code/qml_transforms_qcut.rst]: Nice, done: 3574c97d18783b0320fa23d11980e2dc5f5ffea7

File: doc/code/qml_transforms_qcut.rst (hunk 5 above)
Comment (trbromley) [reply doc/code/qml_transforms_qcut.rst]: Thanks, done!

**Pull request merged**